### PR TITLE
New Presences [BBC, Anime Planet, DramaCool, South Park] Updated Presences [iQIYI, Line TV, Viu, Radiooooo]

### LINKED ISSUE #2787: DramaCool | dramacool.so
**Prerequisites and essential questions** 
- [x] Is it a popular site?
- [x] Is the website older than 2 months? 
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**makalee#0606** 

**https://dramacool.so/** 

**Display what you're watching and episode you're on (time-lapse)** 

**https://gyazo.com/4246d5992dbeda7955103749d5f25541**

### LINKED ISSUE #3138: Anime Planet | anime-planet.com
**Prerequisites and essential questions** 
- [x] Is it a popular site?
- [x] Is the website older than 2 months? 
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?** 
Noah#4986

**What's the name of the service?** 
www.anime-planet.com | Anime Planet

**What should the Presence display?** 
Anime Planet allows users to watch anime through its website via anime streaming services such as Crunchyroll and HIDIVE, so it if it could display watching anime like those services, that would be great. Anime Planet also allows users to rate anime, and put anime on "currently watching," "wanting to watch," "dropped," lists, similar to MyAnimeList and Anilist, so it should display similar to what the presences for those websites display.

**If possible, please provide a logo for the service (512x512 minimum)** 
DzJ9UFDUwAAKj9C

### LINKED ISSUE #3236: South Park | southpark.lat
**Prerequisites and essential questions** 
- [x] Is it a popular site?
- [x] Is the website older than 2 months? 
- [x] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [x] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?** ! 𝑷𝒊𝒌𝒂#8280

**What's the name of the service?**  South Park https://www.southpark.lat/

**What should the Presence display?**  The name of the chapter, season and episode. Same as Netflix

**If possible, please provide a logo for the service (512x512 minimum)** ]
unnamed

### PR DESCRIPTION
**Pull request by SlowLife1661**
# Info
- **Transfer `BBC iPlayer` to myself and changed the service name to `BBC`**
  - [x] BBC iPlayer
  - [x] BBC Sounds 
  - [x] BBC News
  - [x] BBC Sport
  - [x] BBC Weather
  - [x] Home Page
- **New Presence `Anime Planet`**
    - Resolves #3138
- **New Presence `DramaCool`**
    - Resolves #2787
- **New Presence `South Park`**
    - Resolves #3236 
- **Small Changes**
    - Update `metadata`
    - Remove types

## Anime Planet
image
image
image

## BBC
image
image
image
image
image
image
image
image
image
image
image
image
image
image
* Might as well add more pages if users were requested kek

## South Park
image
image
image
image

## DramaCool
image
image
image

## iQIYI
-  New setting `logo`
- Support `more pathname` not all cuz lazy

image
image
image
image
image

### CODE REVIEW

**Review by StrikerFRFX — COMMENTED** [review]
I'm pretty sure that @trains2050 mentioned to me that there's not a presence for the BBC website... could add that as well as Player functionality since some articles have videos which is the same player as iPlayer afaik (or the BBC website has a link to iPlayer within it idk don't use BBC website kek).

**Comment by SlowLife1661** [comment]
👍

**Comment by SlowLife1661** [comment]
Closing this untill BBC is done

**Comment by trains2050** [comment]
@SlowLife1661 
these services should be added 
1. BBC news 
2. bbc sounds 
3. bbc iplayer 
4. the bbc home page 
5. weather 
6. sport

**Comment by SlowLife1661** [comment]
Alright thanks for those

**Comment by StrikerFRFX** [comment]
Stop reopening and closing as if it's a draft, reopen once you've finished your changes and edit your screenshots.

**Comment by SlowLife1661** [comment]
I mean I missclick kek

**Comment by StrikerFRFX** [comment]
> I mean I missclick kek

mkay couldn't be me kek

**Comment by SlowLife1661** [comment]
Sport and news is done sceenshots updated

**Comment by trains2050** [comment]
> Sport and news is done sceenshots updated, some pages may be missing because yes

Reopen when you are ready

**Comment by SlowLife1661** [comment]
I will get this done *by Sunday*, I was quite busy at the moment.

File: websites/A/Anime Planet/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.0",
```
Comment (Bas950) [inline websites/A/Anime Planet/dist/metadata.json:2]: use latest schema (v1.3)

File: websites/A/Anime Planet/presence.ts [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+
+        presenceData.buttons = [
+          {
+            label: (await strings).viewEpisode,
+            url: document.baseURI
+          },
+          {
+            label: (await strings).viewSeries,
+            url: (document.querySelector('h2.sub > a') as unknown as EHref).href
+          }
```
Comment (Bas950) [inline websites/A/Anime Planet/presence.ts:218]: as unknown needed?

File: websites/B/BBC/dist/metadata.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.0",
+  "author": {
+    "name": "SlowLife",
+    "id": "374905512661221377"
+  },
+  "service": "BBC",
+  "description": {
+    "en": "The best of the BBC, with the latest news and sport headlines, weather, TV & radio highlights and much more from across the whole of BBC Online."
+  },
```
Comment (Bas950) [inline websites/B/BBC/dist/metadata.json:10]: add the authors / contributors of the orginal bbc presence as contributor

File: websites/B/BBC/presence.ts [hunk 4]
```diff
@@ -0,0 +1,780 @@
...
+        presenceData.state = title;
+      } else if (path.includes("/sport/formula1/")){
+        presenceData.details = (await strings).readingAricle;
+        presenceData.state = title;
+        presenceData.buttons = [
+          {
+            label: "Read",
+            url: document.baseURI
+          }
+        ];
```
Comment (Bas950) [inline websites/B/BBC/presence.ts:183]: Read? Do you mean Read Article?

File: websites/B/BBC/presence.ts [hunk 5]
```diff
@@ -0,0 +1,780 @@
...
+      } else if (path.includes("/formula1/")){
+        presenceData.details = (await strings).readingAricle;
+        presenceData.state = title;
+        presenceData.buttons = [
+          {
+            label: "Read",
+            url: document.baseURI
+          }
```
Comment (Bas950) [inline websites/B/BBC/presence.ts:191]: same here

File: websites/B/BBC/presence.ts [hunk 6]
```diff
@@ -0,0 +1,780 @@
...
+        ];
+      } else if (path.includes("/transfers")){
+        presenceData.details = (await strings).readingAricle;
+        presenceData.state = title;
+        presenceData.buttons = [
+          {
+            label: "Read",
+            url: document.baseURI
```
Comment (Bas950) [inline websites/B/BBC/presence.ts:219]: here 2

File: websites/B/BBC/presence.ts [hunk 7]
```diff
@@ -0,0 +1,780 @@
...
+        presenceData.details = (await strings).viewPage;
+        presenceData.state = "European's Cricket";
+      } else if (path.includes("/football/")){
+        presenceData.details = (await strings).readingAricle;
+        presenceData.state = title;
+        presenceData.buttons = [
+          {
+            label: "Read",
+            url: document.baseURI
+          }
```
Comment (Bas950) [inline websites/B/BBC/presence.ts:241]: again

File: websites/B/BBC/presence.ts [hunk 8]
```diff
@@ -0,0 +1,780 @@
...
+      } else if (path.includes("/rugby-union/")){
+        presenceData.details = (await strings).readingAricle;
+        presenceData.state = title;
+        presenceData.buttons = [
+          {
+            label: "Read",
+            url: document.baseURI
+          }
+        ];
+      }
```
Comment (Bas950) [inline websites/B/BBC/presence.ts:305]: more

File: websites/B/BBC/presence.ts [hunk 9]
```diff
@@ -0,0 +1,780 @@
...
+    } else if (path.includes("/sport/")){
+      presenceData.details = (await strings).readingAricle;
+      presenceData.state = title;
+      presenceData.buttons = [
+        {
+          label: "Read",
+          url: document.baseURI
+        }
```
Comment (Bas950) [inline websites/B/BBC/presence.ts:367]: everywhere?

File: websites/B/BBC/presence.ts [hunk 10]
```diff
@@ -0,0 +1,780 @@
...
+        smallImageKey: "map"
+      },
+      "/weather/([0-9])": {
+        details: "Viewing weather of:",
+        state: location,
+        smallImageText: weather,
+        smallImageKey: "reading",
+        buttons: [{
+          label: "View",
+          url: document.baseURI
```
Comment (Bas950) [inline websites/B/BBC/presence.ts:464]: View what?

File: websites/I/iQIYI/presence.ts [hunk 11]
```diff
@@ -28,20 +17,14 @@ const presence = new Presence({
...
-      smallImageKey: "search",
-      smallImageText: (await strings).browse,
-      startTimestamp: browsingStamp
-    },
-    newLang = await presence.getSetting("lang"),
+  const newLang = await presence.getSetting("lang"),
     showButtons: boolean = await presence.getSetting("buttons"),
-    searchQuery: boolean = await presence.getSetting("searchQuery");
+    searchQuery: boolean = await presence.getSetting("searchQuery"),
+    logo = ["iqiyi_logo", "iqiyi_logo_b"][await presence.getSetting("logo")];
```
Comment (Bas950) [inline websites/I/iQIYI/presence.ts:28]: you could just set this as the largeImageKey, doesnt needs its own const

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/D/DramaCool/dist/metadata.json [hunk 12]
```diff
@@ -0,0 +1,32 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.3",
+    "author": {
+      "name": "SlowLife",
+      "id": "374905512661221377"
+    },
+    "url": "dramacool.vc",
```
Comment (Bas950) [inline websites/D/DramaCool/dist/metadata.json:7]: please use regex, since they change urls a lot

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Review by Bas950 — DISMISSED** [review]

**Review by Bas950 — CHANGES_REQUESTED** [review]
upload new proof of these latest changes working

**Comment by SlowLife1661** [comment]
See above reee

**Review by Bas950 — APPROVED** [review]

**Review by StrikerFRFX — APPROVED** [review]

**Pull request merged**